packages/@react-aria/utils/src/runAfterTransition.ts

```diff
@@ -91,9 +91,23 @@ if (typeof document !== 'undefined') {
   }
 }
 
+/**
+ * Cleans up any elements that are no longer in the document.
+ * This is necessary because we can't rely on transitionend events to fire
+ * for elements that are removed from the document while transitioning.
+ */
+function cleanupDetachedElements() {
+  for (const [element] of transitionsByElement) {
+    if (element instanceof HTMLElement && !document.contains(element)) {
```

Comment:
using isConnected can help us avoid some odd edgecases and should theoretically be faster as well https://developer.mozilla.org/en-US/docs/Web/API/Node/isConnected
```suggestion
    if (element instanceof HTMLElement && !element.isConnected) {
```

Reply:
TIL. Updated.